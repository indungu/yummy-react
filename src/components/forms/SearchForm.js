import React from 'react';

export default (props) => {
  return (
    <div
      className="col-md-10"
    >
      <form
        className="form-row align-items-center"
        onSubmit={props.onSubmit}
      >
        <div className="form-group mx-sm-3 mb-2 col-9">
          <input
            type="text"
            className="form-control"
            placeholder={props.scope}
            name="queryString"
            onChange={props.onChange}
          />
        </div>
        <button type="submit" className="btn btn-primary mb-2">Search</button>
      </form>
    </div>
  );
};
