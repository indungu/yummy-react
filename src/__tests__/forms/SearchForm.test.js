import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SearchForm from '../../components/forms/SearchForm';

describe('Search form', () => {
  const wrapper = shallow(<SearchForm />);
  it('Renders properly without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
