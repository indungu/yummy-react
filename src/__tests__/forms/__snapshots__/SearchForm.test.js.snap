// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search form Renders properly without crashing 1`] = `
<div
  className="col-md-10"
>
  <form
    className="form-row align-items-center"
  >
    <div
      className="form-group mx-sm-3 mb-2 col-9"
    >
      <input
        className="form-control"
        name="queryString"
        type="text"
      />
    </div>
    <button
      className="btn btn-primary mb-2"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
